### PR TITLE
Removed _gc methods with yields

### PR DESCRIPTION
**Pull request by RepentantGopher**
Removed _gc methods with yields from producer and consumer and fixed hanging forever on close / destroy consumer / producer

### CODE REVIEW

File: kafka/tnt_kafka.c [hunk 1]
```diff
@@ -25,8 +25,8 @@ luaopen_kafka_tntkafka(lua_State *L) {
             {"poll_rebalances", lua_consumer_poll_rebalances},
             {"store_offset", lua_consumer_store_offset},
             {"close", lua_consumer_close},
+            {"destroy", lua_consumer_destroy},
```
Comment (olegrok) [inline kafka/tnt_kafka.c:28]: We shouldn't do that silently.
Currently it's OK, but please file an issue ("implement __gc metamethod for producers and consumers") and add it to list of "known issues".

Also please reflect this changes in readme example.

File: tests/consumer.lua [hunk 2]
```diff
@@ -100,6 +100,8 @@ local function consume(timeout)
                 if err ~= nil then
                     log.error("got error '%s' while commiting msg from topic '%s'", err, msg:topic())
                 end
+            else
```
Comment (olegrok) [inline tests/consumer.lua:103]: I don't see "producer:destroy()" and "consumer:destroy()" calls in tests. How did you check it works?

Comment (olegrok) [reply tests/consumer.lua:103]: My bad. It was done in init.lua

**Review by olegrok — APPROVED** [review]

File: tests/consumer.lua [hunk 3]
```diff
@@ -128,6 +130,8 @@ end
 local function close()
     log.info("closing consumer")
     local exists, err = consumer:close()
+    print(exists)
+    print(err)
```
Comment (RunsFor) [inline tests/consumer.lua:134]: redundant prints here

**Pull request merged**